### DISCUSSION TITLE
'List' instead of 'Array'?

### DISCUSSION THREAD

**Discussion by GeorgeAzma** [carbon-language/carbon-lang#1967] [Language design]

**Comment by SamuelTrew** [comment]
An array is typically of a constant size as memory is allocated for it once and that is it. 

Lists (at least in higher level languages) typically dynamically allocate their memory, allowing for growing and shrinking of list size.

Assuming Carbon arrays follow this style it would not make sense to call it a list

**Reply by DevNoteHQ** [reply]
Also it's pretty much guaranteed that Carbon will add a List-Type later on.
